src/signers/LightwalletSigner.ts

```diff
@@ -0,0 +1,87 @@
+
+import { TxSigner } from './TxSigner'
+import { User } from '../User'
+import { Utils } from '../Utils'
+import {
+  TxInfos,
+  TxInfosRaw,
+  RawTxObject
+} from '../typings'
+
+import { BigNumber } from 'bignumber.js'
+import * as lightwallet from 'eth-lightwallet'
+// declare let lightwallet
+
+/**
+ * The LightwalletSigner class contains functions for signing transactions with eth-lightwallet.
+ */
+export class LightwalletSigner implements TxSigner {
+  private _user: User
+  private _utils: Utils
+
+  constructor (user: User, utils: Utils) {
+    this._user = user
+    this._utils = utils
+  }
+
+  /**
+   * Takes a raw transaction object, turns it into a RLP encoded hex string, signs it with
+   * the loaded user and relays the transaction.
+   * @param rawTx Raw transaction object.
+   */
+  public async confirm (rawTx: RawTxObject): Promise<string> {
+    let rlpTx
+    const txOptions = {
+      ...rawTx,
+      from: rawTx.from.toLowerCase(),
+      gasPrice: this._utils.convertToHexString(rawTx.gasPrice),
+      gasLimit: this._utils.convertToHexString(rawTx.gasLimit),
+      value: this._utils.convertToHexString(rawTx.value)
+    }
+    if (txOptions.to) {
+      txOptions.to = txOptions.to.toLowerCase()
+    }
+    if (rawTx.functionCallData) {
+      rlpTx = lightwallet.txutils.functionTx(
+        rawTx.functionCallData.abi,
+        rawTx.functionCallData.functionName,
+        rawTx.functionCallData.args,
+        txOptions
+      )
+    } else {
+      rlpTx = lightwallet.txutils.valueTx(txOptions)
+    }
+    const signedTx = await this._user.signTx(rlpTx)
```

Comment:
Asking back to user here does not make sense. This is the signing service, so it should do the signing.